ambry-api/src/main/java/com/github/ambry/rest/RestServiceErrorCode.java

```diff
@@ -199,8 +204,12 @@ public static RestServiceErrorCode getRestServiceErrorCode(RouterErrorCode route
         return ServiceUnavailable;
       case InsufficientCapacity:
         return InsufficientCapacity;
+      case BlobUndeleted:
+      case BlobNotDeleted:
+        return Conflict;
       case UnexpectedInternalError:
       case ChannelClosed:
+      case LifeVersionConflict:
```

Comment:
Why is LifeVersionConflict counted as an InternalServerError? does this error code mean that the operation has to be retried after replication converges?

Reply:
First of all it's not a client error for sure. If we see this, then something is wrong with delete and undelete operations. Maybe we have some delete operations performed between undeletes. I would not recommend a retry if we see this, we should try to figure out what happened first.

Reply:
Thanks for the explanation. I thought that it would equate to a conflict http status at first

Reply:
Add some comments to explain why it's `InternalServerError` ?